core/src/main/java/org/apache/calcite/rel/rules/ProjectWindowTransposeRule.java

```diff
@@ -217,6 +217,15 @@ private static ImmutableBitSet findReference(final Project project,
         }
       }
 
+      List<RexNode> analyze = new ArrayList<>();
+      if (group.lowerBound.getOffset() != null) {
+        analyze.add(group.lowerBound.getOffset());
+      }
+      if (group.upperBound.getOffset() != null) {
+        analyze.add(group.upperBound.getOffset());
+      }
+      referenceFinder.apply(analyze);
+
```

Comment:
It doesn't work, since the RexWindowBound does not derive from RexNode.

Reply:
This seems to work in this case.

The style of visitors I am used to requires always the entry point to be a visitor `apply` function, and not an `accept` function of the visited object. This is because the visitor may setup and teardown some data structures on `apply`. 

A quick look through the codebase shows lots of accept calls, so I will use your suggestion.

Reply:
Because this is such a small change I have updated my previous commit instead of creating a new one.